web/models/attendance/attendance_helper.ex

```diff
@@ -84,4 +84,110 @@ defmodule CoursePlanner.AttendanceHelper do
 
     Repo.transaction(updated_multi)
   end
+
+  def update_class_attendance_records(offered_course_id) do
+    query = from oc in OfferedCourse,
+    join: s in assoc(oc, :students),
+    join: c in assoc(oc, :classes),
+    join: cs in assoc(c, :students),
+    preload: [students: s, classes: {c, students: cs}]
+
+    offered_course_with_classes     = Repo.get(query, offered_course_id)
+    offered_course_with_attendances = get_course_attendances(offered_course_id)
+
+    cond do
```

Comment:
This would be better with case and pattern matching, like:
```elixir
  case {att, cls} do
    {nil, nil} -> {:ok, nil}
    {nil, cls} -> ...
    {att, _}   -> ...
```